### PR TITLE
feat: use tabbable shadow dom support

### PR DESCRIPTION
**Pull request by scomea**
## 📖 Description
This pr updates the dialog and toolbar elements to use [tabbable's](https://github.com/focus-trap/tabbable?msclkid=bb1d0c7fcd5b11ec81a99b1d2d3a9767) new shadom dom support.

### 🎫 Issues
Removes some local code.

## 📑 Test Plan

- todo: add some shadow dom tests for dialog/toolbar, ideally in play wright.

### General
- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific
- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/web-components/fast-foundation/src/dialog/dialog.ts [hunk 1]
```diff
@@ -196,28 +196,32 @@ export class Dialog extends FoundationElement {
...
+            },
+        });
     };
 
     /**
      * focus on first element of tab queue
      */
     private focusFirstElement = (): void => {
-        const bounds: (HTMLElement | SVGElement)[] = this.getTabQueueBounds();
+        const bounds: FocusableElement[] = this.getTabQueueBounds();
```
Comment (robarbms) [inline packages/web-components/fast-foundation/src/dialog/dialog.ts:224]: bounds in handleTabKeyDown() should be FocusableElement[] to follow this pattern

**Review by robarbms — APPROVED** [review]

**Comment by stefcameron** [comment]
FYI, tabbable v5.3.0 and focus-trap v6.8.0 with Shadow DOM support have been published to NPM. I hope it helps for this!

**Comment by chrisdholt** [comment]
@scomea let's get this up to date, I'd like to see if we can get this in for the final versions of fast-components.

**Comment by janechu** [comment]
Due to draft status, conflicts, and potential issues with this PR being fairly out of date we're closing as part of #6951.

**Pull request closed without merging**